pkg/executor/adapter.go

```diff
@@ -598,6 +598,7 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 
 	if a.isSelectForUpdate {
 		if sctx.GetSessionVars().UseLowResolutionTSO() {
+			terror.Log(exec.Close(e))
```

Comment:
Should the `exec.Close(e)` also be called at L618 before returning errors?

Reply:
Yes, it should be.
Updated.

Reply:
Or if the `exec.Close(e)` is idempotent, is it better to add the logic in the defer block of the function, if the executor is opened.

Reply:
Let's just do the ad-hoc fix for the bug first here.
I'm afraid of introducing bugs when doing too much changes, although it may reduce potential bug and reduce the maintance burden.
@lcwangchao is doing the refactoring work, I'd like hear his idea

Reply:
Seems there is no better way to make the close more elegant and keep a minimum change of the code. The following function `handlePessimisticSelectForUpdate` and `handleNoDelay` may close or replace the executor internally. If we want to make the code more clear, we should redesign such implementations, introducing more assumption or make `executor.Close` idempotent...

Reply:
@tiancaiamao @lcwangchao 
So we merge this fix first and file an refactor or enhancment issue for the more elegant solution?

Reply:
agree!